homeassistant/components/xiaomi_aqara.py

```diff
@@ -216,13 +232,37 @@ def device_state_attributes(self):
         """Return the state attributes."""
         return self._device_state_attributes
 
-    def push_data(self, data):
+    @callback
+    def _set_unavailable(self, now):
+        """Set state to UNAVAILABLE."""
+        self._remove_unavailability_tracker = None
+        self._is_available = False
+        self.schedule_update_ha_state()
+
+    def _track_unavailable(self):
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
```

Comment:
Make sure that unsubscribing a listener is done in the same type of context (sync/async) as that the listener was added.

Reply:
Now `_track_unavailable()` it's always running in the event loop, via `async_added_to_hass() -> _track_unavailable()` or `async_add_job() -> push_data() -> _track_unavailable()`.